### PR TITLE
RHCLOUD-17416: Update Pre-Flight check to support Sat-RHC

### PR DESCRIPTION
**Pull request by aleccohan**
# What?
Update the Remediations pre-flight check to support the Satellite-RHC use case.

# Why?
With the recent initiative for the shift from receptor to RHC we need to adapt Remediaions to now support 3 main use cases instead of the previous two:

1. Satellite (Receptor)
2. Satellite (RHC)
3. RHC (Direct Connected Systems)

# How?
This change has been implemented through the following steps:

1.  Add `/recipient/status` endpoint to the playbook-dispatcher connector to get the status of rhc_clients.
2. Added mock impl for `/recipient/status` for testing purposes
3. Add `/get_rhc_connections` endpoint to sources connector to get the `rhc_client_id` for Satellite-RHC executors.
4. Add `org_id` param from identity header for dispatcher requests
5. Added logic in fifi.js for the Sat-RHC use case without upsetting the old logic for the two others
6. Change logic for Direct Connect to obtain `rhc_client` connection status from dispatcher
7. Refactor old logic for Receptor & Direct Connect to be more contained and easier to follow
8. Add Sat_org_id to `getConnectionStatus()` endpoint result for later internal use

# Testing?

1.  Updated testing for fifi.js to reflect changes
2. Added testing for `/recipient/status` endpoint
3. Added testing for `get_rhc_connections` endpoint
4. Update testing for fifi.unit.js to reflect changes, deprecate older tests that are no longer needed.
5. Updated scenario testing to be more efficient with the now greater number of results.

# Anything Else?

JIRA: https://issues.redhat.com/browse/RHCLOUD-17416

Secure Coding Practices Checklist Link

    https://github.com/RedHatInsights/secure-coding-checklist
    
    
    ## Secure Coding Practices Checklist GitHub Link
    - https://github.com/RedHatInsights/secure-coding-checklist

    ## Secure Coding Checklist
    - [ ] Input Validation
    - [ ] Output Encoding
    - [ ] Authentication and Password Management
    - [ ] Session Management
    - [ ] Access Control
    - [ ] Cryptographic Practices
    - [ ] Error Handling and Logging
    - [ ] Data Protection
    - [ ] Communication Security
    - [ ] System Configuration
    - [ ] Database Security
    - [ ] File Management
    - [ ] Memory Management
    - [ ] General Coding Practices

### CODE REVIEW

File: src/remediations/fifi.js [hunk 1]
```diff
@@ -428,6 +494,41 @@ function normalize (satellites, smart_management) {
...
+        org_id: systems[0].satelliteOrgId,
+        version: systems[0].satelliteVersion,
+        // unique by ansible host i.e. if there are two systems with the same ansible identifier then
+        // only pick on one of them as we wouldn't be able to tell them apart based on responses from Satellite
+        systems: _(systems).sortBy('id').uniqBy(generator.systemToHost).value()
+    })).values().value();
+
+    await P.map(rhcSatellites, async satellite => {
+        satellite.type = 'RHC-satellite';
+        satellite.source = await sources.getRHCConnections(satellite.id);
```
Comment (jharting) [inline src/remediations/fifi.js:524]: You already fetched the id in `fetchSatRHCClientId`, no? Is it possible to reuse that info?

File: src/connectors/dispatcher/impl.js [hunk 2]
```diff
@@ -23,6 +23,7 @@ module.exports = new class extends Connector {
         this.postRunRequests = metrics.createConnectorMetric(this.getName(), 'postPlaybookRunRequests');
         this.fetchRuns = metrics.createConnectorMetric(this.getName(), 'fetchPlaybookRuns');
         this.fetchRunHosts = metrics.createConnectorMetric(this.getName(), 'fetchPlaybookRunHosts');
+        this.getRunRecipientStatus = metrics.createConnectorMetric(this.getName(), 'getgetPlaybookRunRecipientStatus');
```
Comment (dehort) [inline src/connectors/dispatcher/impl.js:26]: There might be an extra "get" here?

File: src/connectors/dispatcher/impl.js [hunk 3]
```diff
@@ -99,5 +100,43 @@ module.exports = new class extends Connector {
...
+        };
+
+        // This header should be sent to the playbook dispatcher for each internal request.
+        if (auth) {
+            options.headers = {
+                Authorization: `PSK ${auth}`
+            };
+        }
+
+        const result = await this.doHttp (options, false, this.postRunRequests);
```
Comment (dehort) [inline src/connectors/dispatcher/impl.js:126]: Could there be a case here were you might need to fetch more than one "page" of data?

**Review by dehort — APPROVED** [review]
Looks good to me!

I left a couple of comments though.

**Pull request merged**